homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,83 @@
+#include "compression.hpp"
+
+compressedImage compressGrayscale(const image& image) {
+    compressedImage result;
+    for (const auto& imageLine : image) {
+        pixelType lastPixelColor;
+        for (bool startLine = true; const auto& pixelColor : imageLine) {
+            if (pixelColor != lastPixelColor || startLine) {
+                result.emplace_back(pixelColor, 1u);
+                lastPixelColor = pixelColor;
+                startLine = false;
+            } else {
+                ++result.back().second;
```

Comment:
or you can use counter :) it will be a little easier to read. When colors are different you inserter counter and start from 1 and so on.